File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        len_array = len(nums)
        if len_array == 1:
            return nums[0]

        req_rep = len_array//2
        dict = {}
        for val in nums:
            if val in dict:
                dict[val] += 1
                if dict[val] > req_rep:
                     return val
            else:
                dict[val] = 1

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {

        std::unordered_map<int,int> freq_dict;
        for (int i = 0; i < nums.size(); i++)
        {
            if (freq_dict.find(nums[i]) != freq_dict.end())
            {
                return nums[i];
            }
            else
            {
                freq_dict[nums[i]] = 1;
            }
        }
        
    }
};
